### ISSUE TITLE
not working with an actual graphql-server

### ISSUE THREAD

**Issue by chungwong** [4Catalyzer/found-scroll#9]
I found `scroll restoration` is not really working with `found-relay@0.3.0-alpha.9`, at the time of writing.

With a fresh clone of `found-relay`, scroll restoration is working without a problem(with a local schema).
However, once I modified it to talk to a graphql-server, for some reasons scrolling didn't get restored.

I tried to apply found-scroll on the standard [relay todo repo](https://github.com/relayjs/relay-examples/tree/master/todo) too, and scroll restoration is not working as well.

I have also prepared a [demo repo](https://github.com/chungwong/found-relay/commit/cc1326dc11eeb293c5e683f49431135f13743e7d to reproduce the problem with minimal changes.

**Comment by taion** [comment]
Works fine for me. What browser/OS/&c.?

**Comment by taion** [comment]
I mean with your example repo.

**Comment by chungwong** [comment]
Debian 9
Chrome Version 62.0.3202.62 (Official Build) (64-bit)
Firefox 52.4.0 (64-bit)

Win7
Chrome Version 62.0.3202.62 (Official Build) (64-bit)

Here is a screen dump showing the problem.  it was particularly not working on "Completed" while it was working on "All" and "Archive" 

![test](https://user-images.githubusercontent.com/1691249/32199645-98c440c6-be21-11e7-824b-2141029ad863.gif)

**Comment by chungwong** [comment]
Just a little bit update on what I found.

1. If the above actions are repeated , say, for `x` times(repeating `Alt ` + `<-` key and `Alt` + `->` for switching between previous and next page), there is a low chance for the scroll position to be restored.

2. Hard coding the `y` value [here](https://github.com/taion/scroll-behavior/blob/v0.9.4/src/index.js#L280) didn't make a difference

I am starting to believe this is not a `found-scroll` problem but a one belongs to `scroll-behavior`

**Comment by mbaquerizo** [comment]
Apologies for my potential ignorance if this is not related but I'm having some issues as well that seem related (I am working with an actual graphql server). Here is my scenario:

- I have a Book Search page that renders a list of books. When clicking on a book I am taken to the Book Details page for that book.
- Usually the Book Details page will not have a scroll bar; that is, all of its content is in view, so I am taken to [0,0] or the top of the page for a book.
- From the book details page:

    1. When I am at [0,0]: When navigating back I am taken to [0,0] and not the position I was in when clicking the book from the results page.
    2. If the book details page does have a scroll bar, when I navigate to that book I am taken to a position lower than [0,0] (usually will be bottom of the page since Search page is much longer).
        - When navigating back from a the position I am at without scrolling, I am taken close, but not exactly to the position I was in when I clicked the book.
        - When I scroll to a position below [0,0] and navigate back I am taken to that position on the results page

This seems like what is going on in the above gif (notice the 'Completed' route has no scroll bar)
What I expect or would be ideal:

- When navigating to a book I should always taken to the top of the screen regardless of previous position. (I have tried using the `shouldUpdateScroll` prop on `ScrollManager` but this seems to reset the scroll position when navigating back as well)
- When navigating back to the results I should be taken to the scroll position I was previously at on the results page.

Maybe I'm being too optimistic here or maybe I'm going about this the wrong way. For reference here is the route config:

```
<Route path="/" Component={AppWrapper} query={RootViewerQuery}>
  <Route Component={BookSearchContainer} query={bookSearchContainerQuery}>
    <Route Component={BookSearch} query={bookSearchQuery} />
    <Route exact path="/book/:primaryISBN" Component={BookDetailsContainer} query={bookDetailsContainerQuery} />
  </Route>
</Route>
```

Router is set up the default way:
```
const render = createRender({});

const Router = new createFarceRouter({
  historyProtocol: new BrowserProtocol(),
  historyMiddlewares: [queryMiddleware],
  routeConfig: routes,

  render: renderArgs => {
    return (
    <ScrollManager renderArgs={renderArgs}>
      {render(renderArgs)}
    </ScrollManager>
  )}
});
```

**Comment by chungwong** [comment]
If you delay the scrolling, say by 100ms, it will for some reasons, solve the problem

https://github.com/taion/scroll-behavior/blob/master/src/index.js#L303-L304
```
setTimeout(function(){
    scrollLeft(element, left);
    scrollTop(element, top);
}
, 100);
```

**Comment by taion** [comment]
Hmm, I wonder if there's some sort of race condition around things rendering. I really haven't seen this at all.

How are you managing loading states in your cases? Are you using a global pending state, or are you handling pending states on the routes?

**Comment by chungwong** [comment]
For my test case(where the GIF from), it is simply a fresh todo repo plus found-scroll.

**Comment by taion** [comment]
Okay, I think I see what's going on here. Thanks for the repro. I'm not sure why I missed it the first time around. I'll have a fix up shortly.

**Comment by taion** [comment]
Thanks for the patience. This is fixed in v0.1.4.
